some fixes for mwdt toolchain

IRISZZW opened the pull request:
- samples: compression: lz4: add TOOLCHAIN_HAS_NEWLIB filter

    samples\compression\lz4 rely on CONFIG_NEWLIB_LIBC=y, let's
    add TOOLCHAIN_HAS_NEWLIB filter for it.

- tests: mem_protest: workaround aggressive optimization

    We have some static variables var, zeroed_var and bss_var
    in mem_partition.c and we only assert the value of them in
    the same file, so the compiler may pre-calculate it in compile
    stage, it's fine usually.
    But for variable zeroed_var (= 20420), we force to put it in bss
    section in link stage, the value will change in bss clean stage, so
    we will get a wrong result.
    Let's add volatile for these variables to disable pre-calculation.

fixes: #36465 #36466

IRISZZW committed:
samples: compression: lz4: add TOOLCHAIN_HAS_NEWLIB filter

samples\compression\lz4 rely on CONFIG_NEWLIB_LIBC=y, let's
add TOOLCHAIN_HAS_NEWLIB filter for it.

Signed-off-by: Watson Zeng <zhiwei@synopsys.com>

ruuddw commented on tests/kernel/mem_protect/mem_protect/src/mem_partition.c:
```diff
@@ -7,9 +7,9 @@
-static K_APP_DMEM(ztest_mem_partition) int var = 1356;
-static K_APP_BMEM(ztest_mem_partition) int zeroed_var = 20420;
-static K_APP_BMEM(ztest_mem_partition) int bss_var;
+static volatile K_APP_DMEM(ztest_mem_partition) int var = 1356;
```
Can you add some comments to indicate why the 'volatile' was added here?
That's to avoid users copy the example and leave the volatile while in their application it might not be required, it is only for this test.

IRISZZW replied:
added.

ruuddw commented:
In general ok, but there are few typo's in the descriptions (complie, chage, disbale).

IRISZZW committed:
tests: mem_protest: workaround aggressive optimization

We have some static variables var, zeroed_var and bss_var
in mem_partition.c and we only assert the value of them in
the same file, so the compiler may pre-calculate it in compile
stage, it's fine usually.
But for variable zeroed_var (= 20420), we force to put it in bss
section in link stage, the value will change in bss clean stage, so
we will get a wrong result.
Let's add volatile for these variables to disable pre-calculation.

Signed-off-by: Watson Zeng <zhiwei@synopsys.com>

IRISZZW commented:
> In general ok, but there are few typo's in the descriptions (complie, chage, disbale).

my fault, fixed, thanks.
